dmn-tracing-quarkus/src/test/java/org/kie/dmn/kogito/quarkus/tracing/LoanEligibilityIT.java

```diff
@@ -57,10 +58,12 @@ public void testEvaluateLoanEligibility() throws InterruptedException {
         try {
             kafkaClient.consume(TRACING_TOPIC_NAME, s -> {
                 LOGGER.info("Received from kafka: {}", s);
-                Optional.ofNullable(CloudEventUtils.decode(s))
-                        .ifPresentOrElse(
-                                cloudEvent -> countDownLatch.countDown(),
-                                () -> LOGGER.error("Error parsing {}", s));
+
+                if (checkDeserialization(s) && isTraceEventComplete(s)) {
+                    countDownLatch.countDown();
+                } else {
+                    LOGGER.error("Trace event is not valid {}", s);
```

Comment:
Why only log in case of error?
```suggestion
                    fail("Trace event is not valid {}", s);
```

Reply:
It should be the same, since the `countDownLatch` is not decremented the test is going to fail anyway. I refactored as you suggested anyway